scripts/common_startup.sh

```diff
@@ -186,7 +186,13 @@ fi
 if [ $FETCH_WHEELS -eq 1 ]; then
     pip install $requirement_args --index-url "${GALAXY_WHEELS_INDEX_URL}" --extra-index-url "${PYPI_INDEX_URL}"
     GALAXY_CONDITIONAL_DEPENDENCIES=$(PYTHONPATH=lib python -c "from __future__ import print_function; import galaxy.dependencies; print('\n'.join(galaxy.dependencies.optional('$GALAXY_CONFIG_FILE')))")
-    [ -z "$GALAXY_CONDITIONAL_DEPENDENCIES" ] || echo "$GALAXY_CONDITIONAL_DEPENDENCIES" | pip install -r /dev/stdin --index-url "${GALAXY_WHEELS_INDEX_URL}" --extra-index-url "${PYPI_INDEX_URL}"
+    if [ -n "$GALAXY_CONDITIONAL_DEPENDENCIES" ]; then
+        if pip list --format=columns | grep "psycopg2[\(\ ]*2.7.3"; then
```

Comment:
`grep -q`

Reply:
Thanks, good idea.

Reply:
This actually somehow make pip blow out on my shell!
(.venv) $ pip list --format=columns | grep "psycopg2[\(\ ]*2.7.3"
Traceback (most recent call last):
  File "/usr/lib/python2.7/logging/__init__.py", line 885, in emit
    self.flush()
  File "/usr/lib/python2.7/logging/__init__.py", line 845, in flush
    self.stream.flush()
IOError: [Errno 32] Broken pipe
Logged from file list.py, line 272
...

Reply:
@nsoranzo Very strange...  Can you 'pip list' at all, manually from the .venv?

Reply:
I see the same thing here.  https://unix.stackexchange.com/questions/305547/broken-pipe-when-grepping-output-but-only-with-i-flag

Reply:
@dannon See edited message above, adding the `-q` causes the error messages. Same behaviour with e.g. `pip show psycopg2 | grep -q '^Version: 2.7.3$'` . Adding the redirect to `/dev/null` works fine for me, go for it!

Reply:
https://github.com/pypa/pip/issues/4170 is the upstream bug, if you're interested.